pkg/subctl/cmd/validate_tunnel.go

```diff
@@ -0,0 +1,186 @@
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	subv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+
+	"github.com/submariner-io/submariner-operator/apis/submariner/v1alpha1"
+)
+
+const (
+	ClientSourcePort = "9898"
+)
+
+var verboseOutput bool
+
+var validateTunnelCmd = &cobra.Command{
+	Use:   "tunnel <localkubeconfig> <remotekubeconfig>",
+	Short: "Validate if firewall allows tunnels on the Gateway node.",
+	Long:  "This command checks whether firewall configuration allows tunnel to be configured on the Gateway nodes.",
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 2 {
+			return fmt.Errorf("two kubeconfigs must be specified")
+		}
+		same, err := compareFiles(args[0], args[1])
+		if err != nil {
+			return err
+		}
+		if same {
+			return fmt.Errorf("kubeconfig file <localkubeconfig> and <remotekubeconfig> need to have" +
+				" a unique content")
+		}
+		return nil
+	},
+	Run: validateTunnelConfig,
+}
+
+func init() {
+	addValidateFWConfigFlags(validateTunnelCmd)
+	validateTunnelCmd.Flags().BoolVar(&verboseOutput, "verbose", false,
+		"produce verbose logs during validation")
+	validateFirewallConfigCmd.AddCommand(validateTunnelCmd)
+}
+
+func validateTunnelConfig(cmd *cobra.Command, args []string) {
+	localCfg, err := getRestConfig(args[0], "")
+	exitOnError("The provided local kubeconfig is invalid", err)
+
+	remoteCfg, err := getRestConfig(args[1], "")
+	exitOnError("The provided remote kubeconfig is invalid", err)
+
+	validateTunnelConfigAcrossClusters(localCfg, remoteCfg)
+	status.End(status.ResultFromMessages())
+}
+
+func validateTunnelConfigAcrossClusters(localCfg, remoteCfg *rest.Config) {
+	lClientSet, err := kubernetes.NewForConfig(localCfg)
+	exitOnError("Error creating API server client:: %s", err)
+
+	submariner := getSubmarinerResource(localCfg)
+	if submariner == nil {
+		exitWithErrorMsg(submMissingMessage)
+	}
+
+	status.Start(fmt.Sprintf("Validating if tunnels can be setup on Gateway node of cluster %q.",
+		submariner.Spec.ClusterID))
+
+	localEndpoint := getEndpointResource(localCfg, submariner.Spec.ClusterID)
+	if localEndpoint == nil {
+		status.QueueWarningMessage("Could not find the local cluster Endpoint")
+		return
+	}
+
+	tunnelPort, err := getTunnelPort(submariner, localEndpoint)
+	if err != nil {
+		return
+	}
+
+	podCommand := fmt.Sprintf("timeout %d tcpdump -ln -nnX -s 100 -c 5 -i any udp and src port %s and dst port %d",
+		validationTimeout, ClientSourcePort, tunnelPort)
+	sPod, err := spawnSnifferPodOnNode(lClientSet, localEndpoint.Spec.Hostname,
+		namespace, podCommand)
+	if err != nil {
+		status.QueueFailureMessage(fmt.Sprintf("Error while spawning the sniffer pod on the GatewayNode: %v", err))
+		return
+	}
+
+	defer sPod.DeletePod()
+	var gatewayPodIP string
+	if !localEndpoint.Spec.NATEnabled {
+		gatewayPodIP = localEndpoint.Spec.PrivateIP
+	} else {
+		gatewayPodIP = localEndpoint.Spec.PublicIP
+	}
```

Comment:
This won't be accurate with the auto-detection mechanism. Perhaps we should try with both the Private & Public IPs.

Reply:
Done